File: symfony/src/Entity/Post.php
<?php
namespace App\Entity;

use App\Repository\PostRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Table(name: "post")]
#[ORM\Entity(repositoryClass: PostRepository::class)]
class Post
{
    #[ORM\Id]
    #[ORM\GeneratedValue(strategy: "IDENTITY")]
    #[ORM\Column(type: "integer")]
    private int $id;

    #[ORM\Column(type: "string", length: 255)]
    #[Assert\Length(min: 3)]
    private string $sujet;

    #[ORM\Column(type: "string", length: 255)]
    #[Assert\Length(min: 10)]
    private string $description;

    #[ORM\Column(type: "integer")]
    private int $nbrJaime = 0;

    #[ORM\Column(type: "string", length: 255, nullable: true, options: ["default" => "NULL"])]
    private ?string $image = null;

    #[ORM\ManyToOne]
#[ORM\JoinColumn(name: 'user_id', referencedColumnName: 'id_user')]
private ?User $user = null;

    #[ORM\Column(type: "datetime", options: ["default" => "CURRENT_TIMESTAMP"])]
    private \DateTimeInterface $dateP;

    #[ORM\Column(type: "string", length: 255)]
    private string $communaute;

    #[ORM\Column(type: "string", length: 255, nullable: true, options: ["default" => "NULL"])]
    private ?string $analysePo = null;

    #[ORM\Column(type: "integer", nullable: true)]
    private ?int $liked = 0;

    #[ORM\Column(type: "integer", nullable: true)]
    private ?int $badlevel = 0;

    #[ORM\OneToMany(targetEntity: Commentaire::class, mappedBy: "post", orphanRemoval: true)]
    private Collection $Commentaires;

    public function __construct()
    {
        $this->Commentaires = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getSujet(): ?string
    {
        return $this->sujet;
    }

    public function setSujet(string $sujet): self
    {
        $this->sujet = $sujet;
        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): self
    {
        $this->description = $description;
        return $this;
    }

    public function getNbrJaime(): ?int
    {
        return $this->nbrJaime;
    }

    public function setNbrJaime(int $nbrJaime): self
    {
        $this->nbrJaime = $nbrJaime;
        return $this;
    }

    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(?string $image): self
    {
        $this->image = $image;
        return $this;
    }

    public function getUser(): ?User
    {
        return $this->user;
    }

    public function setUser(?User $user): self
    {
        $this->user = $user;

        return $this;
    }

    public function getDateP(): ?\DateTimeInterface
    {
        return $this->dateP;
    }

    public function setDateP(\DateTimeInterface $dateP): self
    {
        $this->dateP = $dateP;

        return $this;
    }

    public function getCommunaute(): ?string
    {
        return $this->communaute;
    }

    public function setCommunaute(string $communaute): self
    {
        $this->communaute = $communaute;

        return $this;
    }

    public function getAnalysePo(): ?string
    {
        return $this->analysePo;
    }

    public function setAnalysePo(?string $analysePo): self
    {
        $this->analysePo = $analysePo;

        return $this;
    }

    public function getLiked(): ?int
    {
        return $this->liked;
    }

    public function setLiked(?int $liked): self
    {
        $this->liked = $liked;

        return $this;
    }

    public function getBadlevel(): ?int
    {
        return $this->badlevel;
    }

    public function setBadlevel(?int $badlevel): self
    {
        $this->badlevel = $badlevel;

        return $this;
    }

    /**
     * @return Collection<int, Commentaire>
     */
    public function getCommentaires(): Collection
    {
        return $this->Commentaires;
    }

    public function addCommentaire(Commentaire $commentaire): self
    {
        if (!$this->Commentaires->contains($commentaire)) {
            $this->Commentaires[] = $commentaire;
            $commentaire->setPost($this);
        }

        return $this;
    }

    public function removeCommentaire(Commentaire $commentaire): self
    {
        if ($this->Commentaires->removeElement($commentaire)) {
            // set the owning side to null (unless already changed)
            if ($commentaire->getPost() === $this) {
                $commentaire->setPost(null);
            }
        }

        return $this;
    }


}

File: mobile/src/com/fancytrade/entities/Post.java
package com.fancytrade.entities;

import com.fancytrade.utils.DateUtils;
import com.fancytrade.utils.Statics;

import java.util.Date;

public class Post implements Comparable<Post> {

    private int id;
    private User user;
    private String sujet;
    private String description;
    private int nbrJaime;
    private String image;
    private Date date;
    private String communaute;
    private String analyse;
    private int liked;
    private int badlevel;

    public Post() {
    }

    public Post(int id, User user, String sujet, String description, int nbrJaime, String image, Date date, String communaute, String analyse, int liked, int badlevel) {
        this.id = id;
        this.user = user;
        this.sujet = sujet;
        this.description = description;
        this.nbrJaime = nbrJaime;
        this.image = image;
        this.date = date;
        this.communaute = communaute;
        this.analyse = analyse;
        this.liked = liked;
        this.badlevel = badlevel;
    }

    public Post(User user, String sujet, String description, int nbrJaime, String image, Date date, String communaute, String analyse, int liked, int badlevel) {
        this.user = user;
        this.sujet = sujet;
        this.description = description;
        this.nbrJaime = nbrJaime;
        this.image = image;
        this.date = date;
        this.communaute = communaute;
        this.analyse = analyse;
        this.liked = liked;
        this.badlevel = badlevel;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public String getSujet() {
        return sujet;
    }

    public void setSujet(String sujet) {
        this.sujet = sujet;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public int getNbrJaime() {
        return nbrJaime;
    }

    public void setNbrJaime(int nbrJaime) {
        this.nbrJaime = nbrJaime;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public String getCommunaute() {
        return communaute;
    }

    public void setCommunaute(String communaute) {
        this.communaute = communaute;
    }

    public String getAnalyse() {
        return analyse;
    }

    public void setAnalyse(String analyse) {
        this.analyse = analyse;
    }

    public int getLiked() {
        return liked;
    }

    public void setLiked(int liked) {
        this.liked = liked;
    }

    public int getBadlevel() {
        return badlevel;
    }

    public void setBadlevel(int badlevel) {
        this.badlevel = badlevel;
    }


    @Override
    public String toString() {
        return "Post : " +
                "id=" + id
                + "\\n User=" + user
                + "\\n Sujet=" + sujet
                + "\\n Description=" + description
                + "\\n NbrJaime=" + nbrJaime
                + "\\n Image=" + image
                + "\\n Date=" + date
                + "\\n Communaute=" + communaute
                + "\\n Analyse=" + analyse
                + "\\n Liked=" + liked
                + "\\n Badlevel=" + badlevel
                ;
    }

    @Override
    public int compareTo(Post post) {
        switch (Statics.compareVar) {
            case "User":
                return this.getUser().getEmail().compareTo(post.getUser().getEmail());
            case "Sujet":
                return this.getSujet().compareTo(post.getSujet());
            case "Description":
                return this.getDescription().compareTo(post.getDescription());
            case "NbrJaime":
                return Integer.compare(this.getNbrJaime(), post.getNbrJaime());
            case "Image":
                return this.getImage().compareTo(post.getImage());
            case "Date":
                DateUtils.compareDates(this.getDate(), post.getDate());
            case "Communaute":
                return this.getCommunaute().compareTo(post.getCommunaute());
            case "Analyse":
                return this.getAnalyse().compareTo(post.getAnalyse());
            case "Liked":
                return Integer.compare(this.getLiked(), post.getLiked());
            case "Badlevel":
                return Integer.compare(this.getBadlevel(), post.getBadlevel());

            default:
                return 0;
        }
    }

}